### DISCUSSION TITLE
Create runtime for ServicePrincipal

### DISCUSSION THREAD

**Discussion by MitaWinata** [microsoft/promptflow#2261] [Q&A]
Is there a way to create runtime with ServicePrincipal? or assign runtime to service principal?

**Comment by D-W-** [comment]
Hi @MitaWinata , you can use compute instance runtime: https://learn.microsoft.com/en-us/azure/machine-learning/prompt-flow/how-to-create-manage-runtime?view=azureml-api-2&tabs=cli%2Cpython#create-a-compute-instance-runtime-on-a-runtime-page. and you can use ServicePricipal to create compute instance.

Also, for Studio UI issue, please refer to [Frequency asked questions (FAQ) — Prompt flow documentation](https://microsoft.github.io/promptflow/cloud/azureai/faq.html) and create feedback on azure portal, thanks!

**Reply by MitaWinata** [reply]
@D-W- , No you cannot do this in UI using Service Principal. That's why this is not a Studio UI issue.

It will be great if creating runtime is supported in PromptFlow SDK/CLI

I found out a better way to do this using AML rest endpoint
